pkg/controller/atlascustomrole/custom_role.go

```diff
@@ -0,0 +1,105 @@
+package atlascustomrole
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translation/customroles"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/workflow"
+)
+
+type roleController struct {
+	ctx       *workflow.Context
+	projectID string
+	service   customroles.CustomRoleService
+	role      *akov2.AtlasCustomRole
+	deleted   bool
+}
+
+func handleCustomRole(ctx *workflow.Context, akoCustomRole *akov2.AtlasCustomRole) workflow.Result {
+	ctx.Log.Debug("starting custom role processing")
+	defer ctx.Log.Debug("finished custom role processing")
+
+	r := roleController{
+		ctx:       ctx,
+		service:   customroles.NewCustomRoles(ctx.SdkClient.CustomDatabaseRolesApi),
+		projectID: akoCustomRole.Spec.ProjectIDRef.ID,
+		deleted:   !akoCustomRole.DeletionTimestamp.IsZero(),
+		role:      akoCustomRole,
+	}
+
+	return r.Reconcile()
+}
+
+func (r *roleController) Reconcile() workflow.Result {
+	currentCustomRoles, err := r.service.List(r.ctx.Context, r.projectID)
```

Comment:
question: we can only list custom roles in Atlas, not get a concrete one?

Reply:
We can call `Get` but this would require to modify the TranslationLayer for the `Get` call (which I think is used somewhere already), as it doesn't return the `httpResponse` which is required in this case to check if role exists.

Reply:
I saw no one is actually calling `GET()` from the sub-resource code. You can modify it to fit your case without problem. Also, I'd return nil and no error when not found, so you know the resource doesn't exist in atlas.